src/main/java/uk/gov/ons/ctp/common/event/EventPublisherImpl.java

```diff
@@ -0,0 +1,86 @@
+package uk.gov.ons.ctp.common.event;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.util.Date;
+import java.util.UUID;
+import lombok.Getter;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.common.event.model.Header;
+import uk.gov.ons.ctp.common.event.model.RespondentAuthenticatedEvent;
+import uk.gov.ons.ctp.common.event.model.RespondentAuthenticatedResponse;
+import uk.gov.ons.ctp.common.event.model.SurveyLaunchedEvent;
+import uk.gov.ons.ctp.common.event.model.SurveyLaunchedResponse;
+
+/** Service responsible for the publication of events. */
+public class EventPublisherImpl implements EventPublisher {
+
+  private static final Logger log = LoggerFactory.getLogger(EventPublisherImpl.class);
+
+  private RabbitTemplate template;
+
+  @Getter
+  public enum EventType {
+    SURVEY_LAUNCHED("RESPONDENT_HOME", "RH"),
+    RESPONDENT_AUTHENTICATED("RESPONDENT_HOME", "RH");
+
+    private String source;
+    private String channel;
+
+    EventType(String source, String channel) {
+      this.source = source;
+      this.channel = channel;
+    }
+  }
+
+  /**
+   * Constructor taking publishing helper class
+   *
+   * @param template Helper class for asynchronous publishing
+   */
+  public EventPublisherImpl(RabbitTemplate template) {
+    this.template = template;
+  }
+
+  /**
+   * Method to publish a respondent Event.
+   *
+   * @param routingKey Message routing key for event
+   * @param payload Message payload for event
+   * @return String UUID transaction Id for event
+   * @throws CTPException
+   */
+  public <T> String sendEvent(String routingKey, T payload) throws CTPException {
```

Comment:
marker interface rather than T please - as it stands it would only fail at runtime if asked to publish Foo
a marker interface would flag at compile time

Reply:
Changed to use marker interface.